src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -357,15 +372,16 @@ private void SetCapacity(int capacity)
             _tail = (_size == capacity) ? 0 : _size;
             _version++;
         }
+        private void MoveNext(ref int index) => MoveNext(ref index, _array);
```

Comment:
Nit: needs a blank line above this. That said, are there many other call sites to this? Why not just use _array at those call sites rather than adding this wrapper?

Reply:
My idea was https://github.com/dotnet/corefx/pull/36502#discussion_r270617794

I see there are other places that could use this too.
* Dequeue
* TryDequeue

By these `array` and `head` is already in registers, so no need to refetch them from memory. That's where `MoveNex(ref, T[])` comes into play.